### PR TITLE
TR: Inject logger into exo-lint

### LINKED ISSUE #124: Inject logger into subclasses of exo commands
Instead of emitting a bunch of events, sometimes through multiple layers, the logger should be injected to classes of the exo commands so that they can call it directly. 

Check off once completed:
- [x] exo-clone
- [x] exo-lint
- [x] exo-sync
- [x] exo-test

### PR DESCRIPTION
**Pull request by trushton**
**Partially** resolves #124 

@kevgo @hugobho

### CODE REVIEW

File: exo-lint/src/cli.ls [hunk 1]
```diff
@@ -11,8 +11,5 @@ module.exports = ->
   app-config = yaml.safe-load fs.read-file-sync('application.yml', 'utf8')
   console.log "Running linter for #{cyan app-config.name}\n"
   logger = new Logger Object.keys(app-config.services)
-  app-linter = new AppLinter app-config
-    ..on 'lint-success', -> logger.log name: 'exo-lint', text: 'Lint passed'
-    ..on 'output', (data) -> logger.log data
-    ..on 'reset-colors', (service-names) -> logger.set-colors service-names
```
Comment (hugoesthere) [inline exo-lint/src/cli.ls:17]: This bit of logic `logger.set-colors service-names` still has to be replaced somewhere else right?

File: exo-lint/src/app-linter.ls [hunk 2]
```diff
@@ -23,16 +22,16 @@ class AppLinter extends EventEmitter
     not-sent = difference Object.keys(received), Object.keys(sent)
 
     if not-received.length is 0 and not-sent.length is 0
-      return @emit 'lint-success'
+      return @logger.log name: 'exo-lint', text: 'Lint passed'
 
     if not-received.length
-      @emit 'output', {name: 'exo lint', text: "The following messages are sent but not received:"}
+      @logger.log {name: 'exo lint', text: "The following messages are sent but not received:"}
```
Comment (kevgo) [inline exo-lint/src/app-linter.ls]: Please remove all these terrible curly braces while you are at it here  :)

File: exo-lint/src/cli.ls (hunk 1 above)
Comment (kevgo) [reply exo-lint/src/cli.ls:17]: +1, I only see usage of `logger.log` in the changes, not `logger.set-colors`.

Comment (trushton) [reply exo-lint/src/cli.ls:17]: The line that was removed is actually unneeded. When initializing the logger with `Object.keys (app-config.services)`, colors are assigned for each of the services. The event that was emitted to reset the colors was passing the same `Object.keys (app-config.services)`, so there was really no reason for it to be there.

**Comment by trushton** [comment]
@kevgo

**Review by kevgo — APPROVED** [review]

**Pull request closed without merging**